src/daemon/collectd.c

```diff
@@ -158,7 +158,7 @@ static int init_global_variables(void) {
   return 0;
 } /* int init_global_variables */
 
-static int change_basedir(const char *orig_dir) {
+static int change_basedir(const char *orig_dir, int nocreate) {
```

Comment:
Please use a positive logic here, "not nocreate" is a double negation and hard to read and I'd much prefer a `create` boolean.

Also, please use `_Bool` instead of `int`.

Reply:
Both done.